### PR TITLE
Complete and correct all Settings calls

### PR DESCRIPTION
**Pull request by shchmue**
I'm not so sure about how the Edid struct is parsed, it has several odd-sized members and the official spec has some dynamic field types, so I went by what was present in my save... Otherwise I tagged all added struct members to the best of my ability but many are unknown.

### CODE REVIEW

File: nx/include/switch/services/set.h [hunk 1]
```diff
@@ -246,13 +551,158 @@ typedef struct {
...
+    u16 unk_x4;
+    u16 unk_x6;
+    u16 unk_x8;
+    u16 unk_xA;
+    u16 unk_xC;
+    u16 unk_xE;
+    u16 unk_x10;
+    u16 unk_x12;
+    u32 cycles;
+} SetSysPtmFuelGaugeParameter;
```
Comment (CTCaer) [inline nx/include/switch/services/set.h]:
```suggestion
typedef struct {
    u16 rcomp0;
    u16 tempc0;
    u16 fullcap;
    u16 fullcapnom;
    u16 lavgempty;
    u16 qresidual00;
    u16 qresidual10;
    u16 qresidual20;
    u16 qresidual30;
    u16 cycles;
    u16 unk0;
    u16 unk1_or_padding;
} SetSysPtmFuelGaugeParameter;
```

File: nx/include/switch/services/fs.h [hunk 2]
```diff
@@ -123,6 +123,11 @@ typedef struct {
     u8 padding[7];
 } FsTimeStampRaw;
 
+/// This is nn::fssystem::ArchiveMacKey. Returned by setsysGetThemeKey. Does not appear to be in use elsewhere.
```
Comment (yellows8) [inline nx/include/switch/services/fs.h]: \ref setsysGetThemeKey

File: nx/include/switch/services/set.h [hunk 3]
```diff
@@ -81,6 +115,12 @@ typedef enum {
     SetSysFriendPresenceOverlayPermission_Friends         = 3,
 } SetSysFriendPresenceOverlayPermission;
 
+typedef enum {
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: Add `/// AudioDevice` before this.

File: nx/include/switch/services/set.h [hunk 4]
```diff
@@ -151,11 +215,108 @@ typedef enum {
     SetSysTouchScreenMode_Standard = 1,   ///< Standard, the default.
 } SetSysTouchScreenMode;
 
+typedef enum {
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: See above.

File: nx/include/switch/services/set.h [hunk 5]
```diff
@@ -151,11 +215,108 @@ typedef enum {
...
+    char name[0x40];
+    Uuid uuid;
+    u32 connection_flags;               ///< Bitmask with \ref SetSysConnectionFlag.
+    u32 wired_flag;
+    u32 connect_to_hidden_network;      ///< Bitmask with UseStealthNetworkFlag.
+    char access_point_ssid[0x20];
+    u32 access_point_ssid_len;
+    u32 access_point_security_type;     ///< Bitmask with \ref SetSysAccessPointSecurityType.
+    u32 access_point_security_standard; ///< Bitmask with \ref SetSysAccessPointSecurityStandard.
+    char access_point_password[0x40];
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: password->passphrase?

File: nx/include/switch/services/set.h [hunk 6]
```diff
@@ -151,11 +215,108 @@ typedef enum {
...
+    char proxy_autoauth_pass[0x20];
+    u16 mtu;
+    u16 padding2;
+} SetSysNetworkSettings;
+
+/// LcdBacklightBrightnessMapping
+typedef struct {
+    float brightness_applied_to_backlight;
+    float ambient_light_sensor_value;
+    float float8;
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: {name}_x{hex}, same below.

File: nx/include/switch/services/set.h [hunk 7]
```diff
@@ -462,6 +929,57 @@ Result setsysGetLockScreenFlag(bool *out);
...
+/**
+ * @brief SetBacklightSettings
+ * @param[in] settings \ref SetSysBacklightSettings
+ */
+Result setsysSetBacklightSettings(const SetSysBacklightSettings *settings);
+
+/**
+ * @brief SetBluetoothDevicesSettings
+ * @param[in] settings Input array of \ref SetSysBluetoothDevicesSettings.
+ * @param[in] count Size of the versions array in entries.
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: settings array

File: nx/include/switch/services/set.h [hunk 8]
```diff
@@ -462,6 +929,57 @@ Result setsysGetLockScreenFlag(bool *out);
...
+ */
+Result setsysSetBluetoothDevicesSettings(const SetSysBluetoothDevicesSettings *settings, s32 count);
+
+/**
+ * @brief GetBluetoothDevicesSettings
+ * @param[out] total_out Total output entries.
+ * @param[out] settings Output array of \ref SetSysBluetoothDevicesSettings.
+ * @param[in] count Size of the versions array in entries.
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: settings array

File: nx/include/switch/services/set.h (hunk 7 above)
Comment (shchmue) [reply nx/include/switch/services/set.h]: ahh looks like copy-paste got me a few places

File: nx/include/switch/services/set.h [hunk 9]
```diff
@@ -474,6 +992,20 @@ Result setsysGetAccountSettings(SetSysAccountSettings *out);
  */
 Result setsysSetAccountSettings(SetSysAccountSettings settings);
 
+/**
+ * @brief GetAudioVolume
+ * @param[out] device \ref SetSysAudioDevice
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: param[in]

File: nx/include/switch/services/set.h [hunk 10]
```diff
@@ -474,6 +992,20 @@ Result setsysGetAccountSettings(SetSysAccountSettings *out);
...
+ * @param[out] out \ref SetSysAudioVolume
+ */
+Result setsysGetAudioVolume(SetSysAudioDevice device, SetSysAudioVolume *out);
+
+/**
+ * @brief SetAudioVolume
+ * @param[in] device \ref SetSysAudioDevice
+ * @param[in] settings \ref SetSysAudioVolume
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: volume

File: nx/include/switch/services/set.h [hunk 11]
```diff
@@ -620,6 +1226,18 @@ Result setsysGetWirelessCertificationFile(void* buffer, size_t size, u64 *out_si
...
 
+/**
+ * @brief GetNetworkSystemClockContext
+ * @param[out] out \ref TimeSystemClockContext
+ */
+Result setsysGetNetworkSystemClockContext(TimeSystemClockContext *out);
+
+/**
+ * @brief SetNetworkSystemClockContext
+ * @param[in] settings \ref TimeSystemClockContext
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: context

File: nx/include/switch/services/set.h [hunk 12]
```diff
@@ -877,6 +1619,44 @@ Result setsysGetHeadphoneVolumeUpdateFlag(bool *out);
...
+ * @note Only available on [3.0.0+].
+ * @param[out] a0 Output arg.
+ * @param[out] a1 Output arg.
+ * @param[out] a2 Output arg.
+ * @param[in] flag Input flag.
+ */
+Result setsysNeedsToUpdateHeadphoneVolume(u8 *a0, u8 *a1, u8 *a2, bool flag);
+
+/**
+ * @brief HetPushNotificationActivityModeOnSleep
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: Get*

File: nx/include/switch/services/set.h [hunk 13]
```diff
@@ -934,11 +1845,49 @@ Result setsysGetRequiresRunRepairTimeReviser(bool *out);
...
+ * @note Only available on [6.0.0+].
+ * @param[in] time_point \ref TimeSteadyClockTimePoint
+ */
+Result setsysSetUserSystemClockAutomaticCorrectionUpdatedTime(const TimeSteadyClockTimePoint *time_point);
+
+/**
+ * @brief GetAccountOnlineStorageSettings
+ * @note Only available on [6.0.0+].
+ * @param[out] total_out Total output entries.
+ * @param[out] settings Output array of \ref SetSysAccountOnlineStorageSettings.
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: Maybe change the param to settings to match this?

File: nx/include/switch/services/set.h [hunk 14]
```diff
@@ -954,6 +1903,92 @@ Result setsysGetPctlReadyFlag(bool *out);
...
+ * @param[out] out \ref SetSysAnalogStickUserCalibration
+ */
+Result setsysGetAnalogStickUserCalibrationR(SetSysAnalogStickUserCalibration *out);
+
+/**
+ * @brief SetAnalogStickUserCalibrationR
+ * @note Only available on [8.1.1+].
+ * @param[in] calibration \ref SetSysAnalogStickUserCalibration
+ */
+Result setsysSetAnalogStickUserCalibrationR(const SetSysAnalogStickUserCalibration *calibration) ;
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: Remove extra space before `;`.

Comment (shchmue) [reply nx/include/switch/services/set.h]: also caught one at setsysSetThemeId and setsysGetPtmBatteryLot

File: nx/include/switch/services/set.h [hunk 15]
```diff
@@ -968,13 +2003,83 @@ Result setsysGetChineseTraditionalInputMethod(SetChineseTraditionalInputMethod *
...
+
+/**
+ * @brief SetZoomFlag
+ * @note Only available on [8.0.0+].
+ * @param[in] flag Input flag.
+ */
+Result setsysSetZoomFlag(bool flag);
+
+/**
+ * @brief On 9.0.0, this is a wrapper for \ref setsysGetPlatFormRegion() == 2.
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: [9.0.0+], and move this into a `@note` (same below).

File: nx/source/services/set.c [hunk 16]
```diff
@@ -657,6 +962,62 @@ Result setsysSetAppletLaunchFlags(u32 flags) {
     return _setCmdInU32NoOut(&g_setsysSrv, flags, 127);
 }
 
+Result setsysGetConsoleSixAxisSensorAccelerationBias(SetSysConsoleSixAxisSensorAccelerationBias *out) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 128, out);
```
Comment (yellows8) [inline nx/source/services/set.c]: *out

File: nx/source/services/set.c [hunk 17]
```diff
@@ -657,6 +962,62 @@ Result setsysSetAppletLaunchFlags(u32 flags) {
...
+}
+
+Result setsysSetConsoleSixAxisSensorAccelerationBias(const SetSysConsoleSixAxisSensorAccelerationBias *bias) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchIn(&g_setsysSrv, 129, bias);
```
Comment (yellows8) [inline nx/source/services/set.c]: *bias

File: nx/source/services/set.c [hunk 18]
```diff
@@ -657,6 +962,62 @@ Result setsysSetAppletLaunchFlags(u32 flags) {
...
+}
+
+Result setsysGetConsoleSixAxisSensorAngularVelocityBias(SetSysConsoleSixAxisSensorAngularVelocityBias *out) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 130, out);
```
Comment (yellows8) [inline nx/source/services/set.c]: *out

File: nx/source/services/set.c [hunk 19]
```diff
@@ -657,6 +962,62 @@ Result setsysSetAppletLaunchFlags(u32 flags) {
...
+}
+
+Result setsysSetConsoleSixAxisSensorAngularVelocityBias(const SetSysConsoleSixAxisSensorAngularVelocityBias *bias) {
+    if (hosversionBefore(4,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchIn(&g_setsysSrv, 131, bias);
```
Comment (yellows8) [inline nx/source/services/set.c]: *bias, likewise below

File: nx/source/services/set.c [hunk 20]
```diff
@@ -674,25 +1035,148 @@ Result setsysSetKeyboardLayout(SetKeyboardLayout layout) {
...
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 145, *out);
+}
+
+Result setsysSetConsoleSixAxisSensorAngularVelocityTimeBias(const SetSysConsoleSixAxisSensorAngularVelocityTimeBias *bias) {
+    if (hosversionBefore(5,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 146, bias);
```
Comment (yellows8) [inline nx/source/services/set.c]: *bias

File: nx/source/services/set.c [hunk 21]
```diff
@@ -674,25 +1035,148 @@ Result setsysSetKeyboardLayout(SetKeyboardLayout layout) {
...
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 147, *out);
+}
+
+Result setsysSetConsoleSixAxisSensorAngularAcceleration(const SetSysConsoleSixAxisSensorAngularAcceleration *acceleration) {
+    if (hosversionBefore(5,0,0))
+        return MAKERESULT(Module_Libnx, LibnxError_IncompatSysVer);
+
+    return serviceDispatchOut(&g_setsysSrv, 148, acceleration);
```
Comment (yellows8) [inline nx/source/services/set.c]: *acceleration

File: nx/source/services/set.c (hunk 17 above)
Comment (shchmue) [reply nx/source/services/set.c]: fixed through cmd 135

File: nx/include/switch/services/set.h (hunk 1 above)
Comment (shchmue) [reply nx/include/switch/services/set.h]: CTC is working on updating this to be more accurate right now, the cycle count fields are wonky

Comment (shchmue) [reply nx/include/switch/services/set.h]: Thanks CTC!

File: nx/include/switch/services/set.h [hunk 22]
```diff
@@ -556,17 +556,17 @@ typedef struct {
...
+    u16 tempc0;
+    u16 fullcap;
+    u16 fullcapnom;
+    u16 lavgempty;
+    u16 qresidual00;
+    u16 qresidual10;
+    u16 qresidual20;
+    u16 qresidual30;
+    u16 cycles;
+    u32 padding;                ///> Normally contains the cycles reg. Unused and contains stack garbage.
```
Comment (yellows8) [inline nx/include/switch/services/set.h]: ///<

File: nx/include/switch/services/fs.h [hunk 23]
```diff
@@ -123,6 +123,11 @@ typedef struct {
     u8 padding[7];
 } FsTimeStampRaw;
 
+/// This is nn::fssystem::ArchiveMacKey. Returned by \ref setsysGetThemeKey. Does not appear to be in use elsewhere.
```
Comment (yellows8) [inline nx/include/switch/services/fs.h]: Mention `\ref setsysSetThemeKey` too.

**Pull request merged**